cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/SystemArgumentsTest.java

```diff
@@ -114,4 +116,23 @@ public void testRetryStrategies() throws InterruptedException {
     // Should give up (returning -1) after passing the max retry time
     Assert.assertEquals(-1L, strategy.nextRetry(1, startTime - 6000));
   }
+
+  @Test
+  public void testLogLevels() {
+    Assert.assertTrue(SystemArguments.getLogLevels(Collections.<String, String>emptyMap()).isEmpty());
+
+    Map<String, String> args = ImmutableMap.of(
+      "system.log.level", "DEBUG",
+      "system.log.level.logger.info", "INFO",
+      "system.log.level.logger.warn", "WARN",
+      "system.log.leveldummyKey", "ERROR"     // <-- This should get picked
+    );
+
+    Map<String, Level> expected = ImmutableMap.of(
+      Logger.ROOT_LOGGER_NAME, Level.DEBUG,
```

Comment:
does this test pass? Given the change above, I would expect it to fail

Reply:
yes it passes. The `system.log.level` should set the ROOT logger level. Any named logger level has to be set with `system.log.level.X`, where `X` is the logger name.